src/mame/konami/konamigq.cpp

```diff
@@ -313,13 +313,13 @@ TIMER_CALLBACK_MEMBER(konamigq_state::scsi_dma_transfer)
 	}
 
 	if (m_dma_requested && m_dma_size > 0)
-		m_dma_timer->adjust(attotime::from_usec(10));
+		m_dma_timer->adjust(attotime::zero);
```

Comment:
Since this is in a timer callback, using a zero delay here will result in the scheduler calling the callback again before executing another timeslice.  You could get more-or-less the same effect with a loop inside this function.

Do you want to leave all the timer adjust calls in and add notes about the timing being unknown, or do you want to change the callbacks to use loops (and maybe still add notes about it)?

FWIW I did try to find information about how DMA timings are generated and got nowhere.

Reply:
I removed the timer adjust from `scsi_dma_transfer` and added a note about finding the proper DMA timings. I think the original situation I was trying to avoid that I have accidentally ran into in the past was a kind of loop situation where scsi_drq would start the DMA transfer, which would then kick off another call to scsi_drq, etc. So the timer calls were used to ensure that the functions would get called after scsi_drq returned so it wouldn't end up in recursion hell. I tested with various konamigv, konamigq, and twinkle games and just using a while loop here works fine.